hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/CockroachLegacyDialect.java

```diff
@@ -465,7 +465,7 @@ public void initializeFunctionRegistry(FunctionContributions functionContributio
 		functionContributions.getFunctionRegistry().register(
 				"trunc",
 				new PostgreSQLTruncFunction(
-						getVersion().isSameOrAfter( 22, 2 ),
+						true,
```

Comment:
This is the legacy dialect which is supposed to still support the old version.

```suggestion
						getVersion().isSameOrAfter( 22, 2 ),
```